src/Bliss/LexiconParser.cc

```
@@ -379,12 +379,90 @@ void LexiconParser::loadWhitelist(const Core::Configuration& config, Core::Strin
+        : LexiconParser(),
+          XmlSchemaParser(c) {
     lexicon_ = _lexicon;
 
     // build schema
     LexiconElement* lexElement = new LexiconElement(this, LexiconElement::creationHandler(&Self::pseudoCreateLexicon), c);
     loadWhitelist(select("vocab"), lexElement->whitelist_);
     setRoot(collect(lexElement));
 }
+
+// use base class parse function
+bool XmlLexiconParser::parseFile(const std::string& filename) {
+    return parser()->Core::XmlSchemaParser::parseFile(filename.c_str()) == 0;
+}
+
+VocabTextLexiconParser::VocabTextLexiconParser(Lexicon* _lexicon)
+        : LexiconParser() {
+    lexicon_          = _lexicon;
+    phonemeInventory_ = new PhonemeInventory();
+}
+
+// parse txt file line by line to a Bliss::Lexicon
+// in the first step, the phonemes are created and the phoneme inventory is set
+// and afterwards the lemmata can be created from these phonemes
+bool VocabTextLexiconParser::parseFile(const std::string& filename) {
+    // collect all labels from the file and add them as phonemes to the phoneme inventory
+    std::ifstream file(filename);
+    if (!file.is_open()) {
+        return false;
+    }
+    std::string line;
+    while (std::getline(file, line)) {
+        if (line.empty())
+            continue;
+        createPhoneme(line);
+    }
+
+    // set the phoneme inventory
+    lexicon_->setPhonemeInventory(Core::ref(phonemeInventory_));
+    // iterate over the phonemes in the inventory to create the lemmata in the lexicon
+    createLemmata();
+    return true;
+}
+
+// helper function to handle one label and create a corresponding phoneme
+void VocabTextLexiconParser::createPhoneme(const std::string& line) {
+    std::string symbol(line);
+    stripWhitespace(symbol);  // in case there are any unintentional whitespaces
+    suppressTrailingBlank(symbol);
+
+    // check if phoneme was already added (if one label appears more than once)
+    if (phonemeInventory_->phoneme(symbol)) {
+        return;
```

In my opinion rasr should not silently accept and skip duplicate symbols. This would shift all indices afterwards which is not expected by the user. I would rather throw an error.